calico_node/startup/startup.go

```diff
@@ -904,6 +915,27 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, cv3 client.Interface) error {
+	_, cv1, err := clients.LoadClients("", "")
+	if err != nil {
+		return err
+	}
+	m := migrator.New(cv3, cv1, nil)
+	if yes, err := m.ShouldMigrate(); err == nil {
+		if yes {
+			_, err = m.Migrate()
```

Comment:
Should have a bit more logging here so that we know if we need migrating / migrating / migrated